src/components/BookStatus/index.tsx

```diff
@@ -10,11 +10,12 @@ import ReservedByYouBookStatus from "./ReservedByYouBookStatus";
 
 type Props = {
   records?: Record[];
+  setStatus?: Function;
 };
 
-const BookStatus = ({ records }: Props) => {
+const BookStatus = ({ records, setStatus }: Props) => {
   const { userProfile } = useAuthContext();
-  const [getBookStatus, setBookStatus] = useState("AVAILABLE");
+  const [getBookStatus, setBookStatus] = useState("");
```

Comment:
Expose calculated status with getBookStatus, and use useEffect in BookDetails to listen for the book status changes and display it. This way we will avoid confusion with setBookStatus and setStatus